workload: seg fault on init tpch

I ran `make bin/workload` followed by `./bin/workload init tpch`, and I got a segfault:

```
Rebeccas-MacBook-Pro:cockroach becca$ ./bin/workload init tpch
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x453411e]

goroutine 55 [running]:
github.com/cockroachdb/cockroach/pkg/workload.Tuples.func1(0xf, 0x0, 0x4, 0x4)
	/Users/becca/go/src/github.com/cockroachdb/cockroach/pkg/workload/workload.go:163 +0x2e
github.com/cockroachdb/cockroach/pkg/workload.Setup.func1(0x2236a2f300000008, 0x4865818)
	/Users/becca/go/src/github.com/cockroachdb/cockroach/pkg/workload/workload.go:318 +0x5a9
github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1(0xc4203f6080, 0xc4203acb90)
	/Users/becca/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:58 +0x57
created by github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go
	/Users/becca/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:55 +0x66
```

I tried the same thing with tpcc and it loaded successfully.

cc @arjunravinarayan, @danhhz

--- a/pkg/workload/workload.go
+++ b/pkg/workload/workload.go
@@ -156,13 +156,16 @@ type BatchedTuples struct {
 
 // Tuples returns a BatchedTuples where each batch has size 1.
 func Tuples(count int, fn func(int) []interface{}) BatchedTuples {
-	return BatchedTuples{
+	t := BatchedTuples{
 		NumBatches: count,
 		NumTotal:   count,
-		Batch: func(batchIdx int) [][]interface{} {
-			return [][]interface{}{fn(batchIdx)}
-		},
 	}
+	if fn != nil {
+		t.Batch = func(batchIdx int) [][]interface{} {
+			return [][]interface{}{fn(batchIdx)}
+		}
+	}
+	return t
 }
 
 // QueryLoad represents some SQL query workload performable on a database
@@ -283,8 +286,8 @@ func Setup(
 	var size int64
 	for _, table := range tables {
 		if table.InitialRows.Batch == nil {
-			// Some workloads don't support initial table data.
-			continue
+			return 0, errors.Errorf(
+				`initial data is not supported for workload %s`, gen.Meta().Name)
 		}
 		batchesPerWorker := table.InitialRows.NumBatches / concurrency
 		g, gCtx := errgroup.WithContext(ctx)

